### PR TITLE
#164368442 Post User Endpoint for Questioner

### PR DESCRIPTION
**Pull request by matthenge**
### What does this PR do?
Enables Users to Create an Account
```
POST /api/auth/signup/
```
### What are the tasks to be completed?
```
- Write tests for user signup
- Ensure Signup is working both basic auth and Google OAuth2 
- Use JWT for basic auth
- Document endpoint in swagger.
```
### How can this be manually tested?
Clone the [repo](https://github.com/Ogutu-Brian/questioners.git):
1. `git checkout ft-signup-authentication-164368442`
## Local Development Setup
 - First Create python virtual env
 ```
  $ virtualenv -p python .venv
 ```
 - Install Requirements
 ```
  $ pip install -r api/requirements.txt
 ```
 - Copy .env-example to .env and set config
 ```
  $ cp .env-example .env
 ```
 - Create postgres database
 ```
  $ sudo su postgres
  $ psql
  postgres=# CREATE USER django WITH PASSWORD 'password';
  postgres=# ALTER ROLE django SET client_encoding TO 'utf8';
  postgres=# ALTER ROLE django SET default_transaction_isolation TO 'read committed';
  postgres=# ALTER ROLE django SET timezone TO 'UTC';
  postgres=# CREATE DATABASE drf;
  postgres=# GRANT ALL PRIVILEGES ON DATABASE drf TO django;
  postgres=# \q
  $ exit
 ```
 - Run Server
 ```
  $ python api/manage.py runserver
 ```
Using postman post post data to `POST  /api/auth/signup/`
### Sample post data
```
{
	"name": "Roberto Rodrigez Olejuera",
        "nick_name": "roberto",
        "email": "baron@yourweb.email",
        "password": "pass456789!"
}
```
### Run tests
```
* python api/manage.py test users
```
### What are the relevant pivotal tracker stories?
[#164368442](https://www.pivotaltracker.com/story/show/164368442)

### CODE REVIEW

File: api/users/tests.py [hunk 1]
```diff
@@ -1,23 +1,126 @@
+"""
```
Comment (davenjeru) [inline api/users/tests.py]: Ignore test files in .codeclimate.yml

**Review by davenjeru — CHANGES_REQUESTED** [review]
Validate that the body contains a valid email:
image

**Comment by Bakley** [comment]
@davenjeru I don't understand the concern about validating email because its already validated to request for a valid email. Could you color more on it
email_val
emai

**Comment by davenjeru** [comment]
@Bakley I had sent a post request without an email and the app crashed. I sent the logs as a screenshot when it crashed.

**Comment by Philipotieno** [comment]
Great work was able to send a confirmation email and activate account :+1:

**Comment by Bakley** [comment]
@davenjeru Fixed the email validation. Writing test. Then will push the changes
email_val

File: api/meetups/tests/test_meetups.py [hunk 2]
```diff
@@ -169,20 +169,7 @@ def force_authenticate_user(self) -> None:
         """
         self.client.force_authenticate(user=self.user)
 
-    def test_successfull_creation_of_meetup(self) -> None:
```
Comment (Ogutu-Brian) [inline api/meetups/tests/test_meetups.py]: Hello @Bakley ,
would you return the test for successful creation of meetup like we agreed then you can just add the rest framework configuration in settings.py like we agreed.

**Review by Ogutu-Brian — CHANGES_REQUESTED** [review]

File: api/meetups/tests/test_meetups.py (hunk 2 above)
Comment (Bakley) [reply api/meetups/tests/test_meetups.py]: Yes, @Ogutu-Brian Done that, I am documenting the endpoint so will add with the changes in the next push

**Review by Amoswachira — APPROVED** [review]
looks good just work on  what @Ogutu-Brian has requested

**Review by IssaIan — COMMENTED** [review]
Try using autopep8. Should contain a line at the end.

**Review by davenjeru — APPROVED** [review]

**Review by IssaIan — APPROVED** [review]

**Review by IssaIan — COMMENTED** [review]
Your build is failing...check on it

**Review by a-braham — APPROVED** [review]
Good job @Bakley. Hell of a job

**Review by Bakley — APPROVED** [review]

**Review by Ogutu-Brian — APPROVED** [review]

**Review by IssaIan — APPROVED** [review]
Great.

**Comment by Ogutu-Brian** [comment]
Good work!

**Pull request merged**